### PR TITLE
CMO-288: Disable Perry HTTP TRACE

### PR DESCRIPTION
**Pull request by david-smith-osi**
### JIRA Issue Link
- [CMO-288: Perry - HTTP TRACE / TRACK Methods Allowed](https://osi-cwds.atlassian.net/browse/CMO-288)

### Tests
- [x] I have included unit tests 
- [ ] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Configuration changes

N/A

### Technical debt

N/A

### Checklist:

- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/config/JettyConfiguration.java [hunk 1]
```diff
@@ -34,15 +43,23 @@ private void customizeJetty(JettyEmbeddedServletContainerFactory container) {
...
+        });
+      }
+
       private void enableHttpPort(JettyEmbeddedServletContainerFactory container) {
         container.addServerCustomizers((JettyServerCustomizer) server -> {
           // HTTP
-          ServerConnector connector = new ServerConnector(server);
+          final ServerConnector connector = new ServerConnector(server);
           connector.setPort(httpPort);
           server.addConnector(connector);
```
Comment (tparkerosi) [inline src/main/java/gov/ca/cwds/config/JettyConfiguration.java]: Jetty or tomcat? Shouldn't the version also be listed on startup?

File: src/main/java/gov/ca/cwds/config/TraceTrackDenialFilter.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.GenericFilterBean;
+
+@Component
+public class TraceTrackDenialFilter extends GenericFilterBean {
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+
```
Comment (tparkerosi) [inline src/main/java/gov/ca/cwds/config/TraceTrackDenialFilter.java]: I recall seeing several solutions and I thought I recall something else also being done. One config was for some warnings. Are you sure this is all?

File: src/main/java/gov/ca/cwds/config/JettyConfiguration.java (hunk 1 above)
Comment (david-smith-osi) [reply src/main/java/gov/ca/cwds/config/JettyConfiguration.java]: I recall that one mode appeared to use Tomcat over Jetty, and so I handled both.
I will remove Tomcat references, if you wish.

File: src/main/java/gov/ca/cwds/config/TraceTrackDenialFilter.java (hunk 2 above)
Comment (david-smith-osi) [reply src/main/java/gov/ca/cwds/config/TraceTrackDenialFilter.java]: Those solutions either extended the Jetty Server class or modified the global servlet context.  It is not clear how to weave those solutions into Spring embedded Jetty.

The selected technique evaluates the HTTP method only after Jetty verifies the web context. Both Perry web contexts, "/perry" and "/manage", are defended from TRACE, and other web contexts result in 404 (not found).

**Pull request merged**